### PR TITLE
Fix MonoCrossAOT.UnixFilePermissions for wasm

### LINKED ISSUE #54612: [mono][workload] Cross compiler pack fails to install on browser due to permission setting
The permission setting fix broke wasm workloads

```
dotnet workload install -v:d microsoft-net-sdk-blazorwebassembly-aot

Skip NuGet package signing validation. NuGet signing validation is not available on Linux or macOS https://aka.ms/workloadskippackagevalidation .
Updated advertising manifest microsoft.net.sdk.tvos.
Updated advertising manifest microsoft.net.sdk.android.
Updated advertising manifest microsoft.net.sdk.maui.
Updated advertising manifest microsoft.net.sdk.macos.
Updated advertising manifest microsoft.net.sdk.maccatalyst.
Updated advertising manifest microsoft.net.sdk.ios.
Updated advertising manifest microsoft.net.workload.mono.toolchain.
Installing workload manifest microsoft.net.sdk.tvos version 15.0.100-ci.upload-download-dotnettests.46.
Installing workload manifest microsoft.net.sdk.android version 30.0.100-preview.6.51.
Installing workload manifest microsoft.net.sdk.maui version 6.0.100-preview.6.874+sha.28d1f1d25-azdo.4904039.
Installing workload manifest microsoft.net.sdk.macos version 12.0.100-ci.upload-download-dotnettests.46.
Installing workload manifest microsoft.net.sdk.maccatalyst version 15.0.100-ci.upload-download-dotnettests.46.
Installing workload manifest microsoft.net.sdk.ios version 15.0.100-ci.upload-download-dotnettests.46.
Installing workload manifest microsoft.net.workload.mono.toolchain version 6.0.0-preview.7.21323.3.
Installing pack Microsoft.NET.Runtime.WebAssembly.Sdk version 6.0.0-preview.7.21323.3...
Writing workload pack installation record for Microsoft.NET.Runtime.WebAssembly.Sdk version 6.0.0-preview.7.21323.3...
Installing pack Microsoft.NETCore.App.Runtime.Mono.browser-wasm version 6.0.0-preview.7.21323.3...
Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.browser-wasm version 6.0.0-preview.7.21323.3...
Installing pack Microsoft.NETCore.App.Runtime.AOT.Cross.browser-wasm version 6.0.0-preview.7.21323.3...
Workload installation failed, rolling back installed packs...
Installing workload manifest microsoft.net.sdk.tvos version 14.5.100-ci.main.723.
Installation roll back failed: Failed to install manifest microsoft.net.sdk.tvos version 14.5.100-ci.main.723: The transaction has aborted..
Rolling back pack Microsoft.NET.Runtime.WebAssembly.Sdk installation...
Uninstalling workload pack Microsoft.NET.Runtime.WebAssembly.Sdk version 6.0.0-preview.7.21323.3.
Rolling back pack Microsoft.NETCore.App.Runtime.Mono.browser-wasm installation...
Uninstalling workload pack Microsoft.NETCore.App.Runtime.Mono.browser-wasm version 6.0.0-preview.7.21323.3.
Rolling back pack Microsoft.NETCore.App.Runtime.AOT.Cross.browser-wasm installation...
Workload installation failed: chmod: /tmp/Microsoft.NETCore.App.Runtime.AOT.Cross.browser-wasm-6.0.0-preview.7.21323.3-extracted/tools/opt: No such file or directory

install
  Install a workload.

Usage:
  dotnet [options] workload install [<WORKLOAD_ID>...]

Arguments:
  <WORKLOAD_ID>  The NuGet Package Id of the workload to install.

Options:
  --sdk-version <VERSION>                                                  The version of the SDK.
  --configfile <FILE>                                                      The NuGet configuration file to use.
  --add-source <SOURCE>                                                    Add an additional NuGet package source to use during installation.
  --skip-manifest-update                                                   Skip updating the workload manifests.
  --from-cache <from-cache>                                                Complete the operation from cache (offline).
  --download-to-cache <download-to-cache>                                  Download packages needed to install a workload to a folder which can be used for offline installation.
  --include-previews                                                       Allow prerelease workload manifests.
  --temp-dir <temp-dir>                                                    Configure the temporary directory used for this command (must be secure).
  --disable-parallel                                                       Prevent restoring multiple projects in parallel.
  --ignore-failed-sources                                                  Treat package source failures as warnings.
  --no-cache                                                               Do not cache packages and http requests.
  --interactive                                                            Allows the command to stop and wait for user input or action (for example to complete authentication).
  -v, --verbosity <d|detailed|diag|diagnostic|m|minimal|n|normal|q|quiet>  Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
  -?, -h, --help                                                           Show help and usage information
```

cc @steveisok @directhex

### PR DESCRIPTION
**Pull request by steveisok**
We introduced Microsoft.NETCore.App.MonoCrossAOT.UnixFilePermissions.xml in https://github.com/dotnet/runtime/pull/54501 to make sure the right permissions are set when installing AOT compiler workload packs.  We hardcoded the list to include mono-aot-cross, llc, and opt.  However, in wasm's case they only have mono-aot-cross.

This change makes the xml file a template and only includes mono-aot-cross for browser and all three for the other configurations.

Fixes https://github.com/dotnet/runtime/issues/54612

### CODE REVIEW

**Comment by steveisok** [comment]
Official build where we need to validate the change.  AOT Compiler packs are only generated here.

https://dev.azure.com/dnceng/internal/_build/results?buildId=1204328&view=results

**Comment by radical** [comment]
This specifies the files to be included, IIUC:
`<NativeRuntimeAsset Include="$(MonoAotCrossDir)$(TargetCrossRid)\**" TargetPath="tools/" />`

Can we get execute permissions from those files, and use that to populate the xml?

**Comment by lewing** [comment]
> This specifies the files to be included, IIUC:
> `<NativeRuntimeAsset Include="$(MonoAotCrossDir)$(TargetCrossRid)\**" TargetPath="tools/" />`
> 
> Can we get execute permissions from those files, and use that to populate the xml?

you could do something like all the files in tools without an extension pretty easily

File: src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj [hunk 1]
```diff
@@ -30,13 +31,7 @@
...
     </ItemGroup>
     <PropertyGroup>
-      <_PermissionsFiles>&lt;File Path=&quot;tools/mono-aot-cross&quot; Permission=&quot;755&quot; /&gt;</_PermissionsFiles>
-    </PropertyGroup>
-    <PropertyGroup Condition="(!$([MSBuild]::IsOsPlatform('Windows')) and '$(TargetCrossRid)' != 'browser-wasm')">
-      <_PermissionsFiles>$(_PermissionsFiles)
-    &lt;File Path=&quot;tools/opt&quot; Permission=&quot;755&quot; /&gt;</_PermissionsFiles>
-      <_PermissionsFiles>$(_PermissionsFiles)
-    &lt;File Path=&quot;tools/llc&quot; Permission=&quot;755&quot; /&gt;</_PermissionsFiles>
+      <_PermissionsFiles>@(_ToolFile -> '&lt;File Path=&quot;tools/%(RecursiveDir)%(FileName)%(Extension)&quot; Permission=&quot;755&quot; /&gt;', ' ')</_PermissionsFiles>
```
Comment (lewing) [inline src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj:34]: I think there might be some json files in there, may need to condition it on %(Extension) == ' '

Comment (steveisok) [reply src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.MonoCrossAOT.sfxproj:34]: It looks like a step in the installer is adding the json files, so we should be good.

**Review by lewing — APPROVED** [review]

**Comment by radical** [comment]
> Official build where we need to validate the change. AOT Compiler packs are only generated here.
> 
> [dev.azure.com/dnceng/internal/_build/results?buildId=1204648&view=results](https://dev.azure.com/dnceng/internal/_build/results?buildId=1204648&view=results)

Will that happen as part of some step there, or do we manually download, and check permissions?

**Review by radical — APPROVED** [review]

**Comment by steveisok** [comment]
> Will that happen as part of some step there, or do we manually download, and check permissions?
Has to be manual for now.

**Comment by steveisok** [comment]
@radical and I validated a sample of the linux and osx packages have the right permissions.  Also, the official build completed.

**Pull request merged**